src/main/java/org/apache/commons/beanutils2/converters/ColorConverter.java

```diff
@@ -0,0 +1,216 @@
+     */
+    public ColorConverter() {
+        super();
+    }
+
+    /**
+     * @param defaultValue The default value to be returned
+     * if the value to be converted is missing or an error
+     * occurs converting the value.
+     */
+    public ColorConverter(final Object defaultValue) {
+        super(defaultValue);
+    }
+
+    @Override
+    protected Class<?> getDefaultType() {
+        return Color.class;
+    }
+
+    /**
+     * <p>
+     *     Convert the configuration value to a Java {@link Color} object,
+     *     by reading the hexadecimal {@link String} and converting each component.
+     * </p>
+     *
+     * <p>
+     *     This can also interpret raw color names based on the standard colors
+     *     defined in Java, such as the following:
+     * </p>
+     *
+     * <ul>
+     *     <li>{@link Color#WHITE}</li>
+     *     <li>{@link Color#LIGHT_GRAY}</li>
+     *     <li>{@link Color#GRAY}</li>
+     *     <li>{@link Color#DARK_GRAY}</li>
+     *     <li>{@link Color#BLACK}</li>
+     *     <li>{@link Color#RED}</li>
+     *     <li>{@link Color#PINK}</li>
+     *     <li>{@link Color#ORANGE}</li>
+     *     <li>{@link Color#YELLOW}</li>
+     *     <li>{@link Color#GREEN}</li>
+     *     <li>{@link Color#MAGENTA}</li>
+     *     <li>{@link Color#CYAN}</li>
+     *     <li>{@link Color#BLUE}</li>
+     * </ul>
+     *
+     * <small>
+     *     Implementation Notes: We specifically avoid the use of {@link Color#decode(String)}
+     *     for hexadecimal {@link String}s starting with {@link #HEX_COLOR_PREFIX}
+     *     as it does not provide the desired result.
+     *     The {@link Color#decode(String)} method uses {@link Integer#decode(String)}
+     *     under the hood to convert the input to a number, which means input like
+     *     <code>#FFF</code> gets interpreted incorrectly as it's literally converted
+     *     to the number <code>0xFFF</code>, rather than the color, <code>#FFFFFF</code> which
+     *     it is short hand for. It also doesn't work for <code>#FFFFFFFF</code> due to it
+     *     being unable to parse as an {@link Integer}.
+     *     If this is desired, then this method falls back to using {@link Color#decode(String)},
+     *     so for literal hexadecimal values you prefix it with <code>0x</code> instead of
+     *     {@link #HEX_COLOR_PREFIX}.
+     * </small>
+     *
+     * @param value The String property value to convert.
+     * @return A {@link Color} which represents the compiled configuration property.
+     * @throws NullPointerException If the value is null.
+     * @throws NumberFormatException If an invalid number is provided.
+     */
+    @Override
+    protected <T> T convertToType(Class<T> type, Object value) throws Throwable {
+        if (Color.class.isAssignableFrom(type)) {
+            Objects.requireNonNull(value, "Value can't be null.");
+            final String stringValue = value.toString();
+
+            switch (stringValue.toLowerCase()) {
```

Comment:
Please let me know if there are any problems with the approach I'll be taking here.

I agree handling `Color#toString()` would be nice. While I was at it, I opted to support some terser versions since I was using regular expression anyway, and it wasn't much additional work:

Regex:
`^(?:[A-Za-z\d._]+)??\[?(?:r=)?(\d{1,3}),(?:g=)?(\d{1,3}),(?:b=)?(\d{1,3})\]?$`

`java.awt.Color[r=255,g=255,b=255]` - The normal `Color#toString()` result.  
`org.apache.ExtendedColor[r=255,g=0,b=255]` - If the `Color` class is extended, but `#toString()` is not overridden.  
`[r=0,g=255,b=0]` - Without the package.  
`r=64,g=64,b=64` - Without the backets.  
`255,175,175` - Without the color labels.